shared/modules/doc/README.md

```diff
@@ -57,36 +163,61 @@ A base module is a module that implements a common interface, exposing the most
+See `ModuleFactoryWithOptions`, also defined in `shared/modules/factory.go`.
 
-Module creation utilizes the factory interfaces defined above.
-Depending on your module's requirements, you can choose the most suitable factory interface for your module (or "sub-module").
+Options stand in contrast to values which are **required** in order for a (sub)module to function properly.
+These values SHOULD be composed in to a single config struct which can be received via the respective factory interface type.
 
-For examples of (sub-)modules that implement `ModuleFactoryWithConfig`, see:
+For modules, this config MUST be incorporated into the [`runtime.Config`](../../../runtime/configs/config.go) which automatically implies the following:
+- such config values can be set via the node's config file and overriden via environment variables
+- such config values MUST be serializable (i.e. no references to objects in memory)
 
-- [`p2pModule`](https://github.com/pokt-network/pocket/tree/main/p2p/module.go) - `WithHostOption`
-- [`rpcCurrentHeightProvider`](https://github.com/pokt-network/pocket/tree/main/p2p/providers/current_height_provider/rpc/provider.go) - `WithCustomRPCURL`
+This serializable config value constraint on modules will likely foster the creation of submodules anywhere modules depend on high-level (non-serializable) objects, thus moving the responsibility of managing those dependencies to the submodule.
 
+For examples of modules and options see:
 
-Essentially the `ModuleOption` sets a custom RPC URL for the module at runtime.
+- `Module`: [`p2pModule`](../../../p2p/module.go)
+- `ModuleOption`: [`WithHost`](../../../p2p/testutil.go)
+- `ModuleOption`: [`WithStakedActorRouter`](../../../p2p/testutil.go)
+- `ModuleOption`: [`WithUnstakedActorRouter`](../../../p2p/testutil.go)
 
+### Submodule creation
+
+Depending on your submodule's requirements, you SHOULD choose the most suitable factory interface.
+
+Options functions used in conjunction with submodules MAY accept either concrete or interface submodule types.
+If an option is only intended or only makes sense to be used with a specific concrete submodule type, then it SHOULD either accept that concrete type or be rewritten to apply to the respective interface type, generally.
+
+Options stand in contrast to values which are **required** in order for a (sub)module to function properly.
+These values SHOULD be composed in to a single config struct which can be received via the respective factory interface type.
+
+Submodules aren't required to conform to the `ModuleFactoryWithOptions` interface like modules are.
+Instead, each submodule SHOULD define its own factory interface using the applicable shared generic `Factory...` (see: [Factory Interfaces](#factory-interfaces)) interface, which it SHOULD also enforce.
+Each submodule factory interface type SHOULD (i.e. prefer to) return an interface type for the submodule, but it MAY be appropriate to return a concrete type in some cases.
```

Comment:
Can you add a TODO to simplify this. 

ALl the documentation is really good, but it turned into a lot of reading w/ lots of rules around MAY/SHOULD/MUST of module/submodule and I feel like there's an opportunity for a deck or leveraging GPT to explain it better.

Even I started getting a bit confused around all the patterns given the context I have, so I know it'll be harder for others.

Reply:
I'll take another pass at it. 🫡